### PR TITLE
LPS-200973 Change property list keyboard navigation in segment editor

### PR DESCRIPTION
**Pull request by claraizquierdo**
Motivation
=======
We want to unify the keyboard navigation with the one in page editor fragment list.
[Link to story or ticket](https://liferay.atlassian.net/browse/LPS-200973)

Steps to Verify:
----------------
1. Go to People -> Segments
1. Click on add new user segment
1. Navigate through the Properties sidebar using the keyboard and check that everything works as describe in the ticket.

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_sidebar/CriteriaSidebarCollapse.js [hunk 1]
```diff
@@ -69,101 +71,113 @@ function filterProperties(properties, searchValue) {
...
+								displayType="secondary"
+								label={filteredProperties.length}
+							/>
+						)}
+					</ClayPanel.Title>
+				</div>
+			}
+			displayType="unstyled"
+			expanded={active}
+			id={key}
```
Comment (sandrodw3) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_sidebar/CriteriaSidebarCollapse.js:121]: Why do we need this id?

**Review by sandrodw3 — COMMENTED** [review]
Hey @claraizquierdo! It's working very well overall 🥳 I just have a couple of comments and suggestions:
- As we are repeating most of the logic from page editor's `useKeyboardNavigation`, would it be possible to extract it to `layout-js-components-web`? And migrate it to typescript? 🤗
- Testing it, I think there's something we have to fix. I see that tabbing, we move from one collapse header to another. I think the correct pattern is to exit the list if we press tab
- You are also expanding and collapsing the panels with left and right arrow indistinctly. I think we should open it with right, and close it with left
- I guess this is still a WIP but just in case, I saw the focus style of the drag handler is black. I also think we shouldn't remove the hover style when focusing the drag handler

image

**Comment by claraizquierdo** [comment]
Hi @sandrodw3 ,

Thank you very much for your review. I'm going to answer your comments inline:

> Hey @claraizquierdo! It's working very well overall 🥳 I just have a couple of comments and suggestions:
> 
> * As we are repeating most of the logic from page editor's `useKeyboardNavigation`, would it be possible to extract it to `layout-js-components-web`? And migrate it to typescript? 🤗

At the beginning I thought about reusing the code, but the way we search for the node we need to move to is very different  and we will need to pass most of the code of `onHeaderKeyDown` and `onListItemKeyDown` functions 😞 

> * Testing it, I think there's something we have to fix. I see that tabbing, we move from one collapse header to another. I think the correct pattern is to exit the list if we press tab

I didn't see that requirement in the story description but I can talk to Carol and check if we should do it this way :)

> * You are also expanding and collapsing the panels with left and right arrow indistinctly. I think we should open it with right, and close it with left

I will try to fix this.

> * I guess this is still a WIP but just in case, I saw the focus style of the drag handler is black. I also think we shouldn't remove the hover style when focusing the drag handler
> 
> image

You are right. I'm going to fix it.

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_sidebar/CriteriaSidebarCollapse.js (hunk 1 above)
Comment (claraizquierdo) [reply modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_sidebar/CriteriaSidebarCollapse.js:121]: 🤔

**Comment by claraizquierdo** [comment]
ci:forward

**Comment by claraizquierdo** [comment]
ci.forward

**Comment by jkappler** [comment]
ci:forward

**Pull request closed without merging**